massa-grpc/src/tests/stream.rs

```diff
@@ -0,0 +1,1580 @@
+use std::{net::SocketAddr, ops::Add, time::Duration};
+
+use crate::tests::mock::{grpc_public_service, MockExecutionCtrl, MockPoolCtrl};
+use massa_consensus_exports::test_exports::MockConsensusControllerImpl;
+use massa_execution_exports::{ExecutionOutput, SlotExecutionOutput};
+use massa_models::{
+    address::Address, block::FilledBlock, secure_share::SecureShareSerializer, slot::Slot,
+    stats::ExecutionStats,
+};
+use massa_proto_rs::massa::{
+    api::v1::{
+        public_service_client::PublicServiceClient, NewBlocksRequest, NewFilledBlocksRequest,
+        NewOperationsRequest, NewSlotExecutionOutputsRequest, SendEndorsementsRequest,
+        SendOperationsRequest, TransactionsThroughputRequest,
+    },
+    model::v1::{Addresses, Slot as ProtoSlot, SlotRange},
+};
+use massa_protocol_exports::{
+    test_exports::tools::{
+        create_block, create_block_with_operations, create_endorsement,
+        create_operation_with_expire_period,
+    },
+    MockProtocolController,
+};
+use massa_serialization::Serializer;
+use massa_signature::KeyPair;
+use massa_time::MassaTime;
+use serial_test::serial;
+use tokio_stream::StreamExt;
+
+// const GRPC_SERVER_URL: &str = "grpc://localhost:8888";
+
+#[tokio::test]
+#[serial]
+async fn transactions_throughput_stream() {
+    let addr: SocketAddr = "[::]:4017".parse().unwrap();
+    let mut public_server = grpc_public_service(&addr);
+    let config = public_server.grpc_config.clone();
+
+    let mut exec_ctrl = MockExecutionCtrl::new();
+
+    exec_ctrl.expect_clone().returning(|| {
```

Comment:
Try to make a small comments in each test to make easily readable: GivenWhenThen